src/http.rs

```diff
@@ -67,8 +67,20 @@ where
     }
 
     /// Make a get request and parse into JSON
-    pub async fn get_json<T: DeserializeOwned>(&self, endpoint: &str) -> Result<T, C::Error> {
-        C::get_json(self.get(endpoint)).await
+    pub async fn get_json<T: DeserializeOwned>(
+        &self,
+        endpoint: &str,
+        interval: Option<u64>,
```

Comment:
decided to inject interval here so that each get_json can decide on the interval.